src/calls/components/Calendar.jsx

```diff
@@ -0,0 +1,314 @@
+        const startDay = getDay(startOfMonth(this.state.selectedDate));
+        const endDay = getDay(endOfMonth(this.state.selectedDate));
+
+        const startOfFirstWeek = startOfWeek(
+            startOfMonth(this.state.selectedDate),
+            {
+                weekStartsOn: 1
+            }
+        ).getDate();
+        const numOfDaysInPrevouosMonth = getDaysInMonth(
+            subMonths(this.state.selectedDate, 1)
+        );
+        const startOfLastWeek = lastDayOfWeek(
+            lastDayOfMonth(this.state.selectedDate),
+            {
+                weekStartsOn: 1
+            }
+        ).getDate();
+
+        let prevMonthDays = [];
+        let days = [];
+        let nextMonthDays = [];
+
+        // Previous
+        for (
+            let i = startOfFirstWeek;
+            i <= numOfDaysInPrevouosMonth && startDay !== 1;
+            i++
+        ) {
+            prevMonthDays.push(
+                <td
+                    type="prev"
+                    key={i + 'p'}
+                    onClick={this.handleOnDateClick}
+                    className="text-center text-secondary"
+                >
+                    {i}
+                </td>
+            );
+        }
+        // Actual
+        const selectedDay = this.state.selectedDate.getDate();
+        for (let i = 1; i <= numOfDays; i++) {
+            if (i === selectedDay) {
+                days.push(
+                    <td
+                        key={i + 'd'}
+                        onClick={this.handleOnDateClick}
+                        className="text-center btn-success"
+                    >
+                        {i}
+                    </td>
+                );
+            } else {
+                days.push(
+                    <td
+                        key={i + 'd'}
+                        onClick={this.handleOnDateClick}
+                        className="text-center"
+                    >
+                        {i}
+                    </td>
+                );
+            }
+        }
+        // Next
+        for (let i = 1; i <= startOfLastWeek && endDay > 0; i++) {
+            nextMonthDays.push(
+                <td
+                    type="next"
+                    key={i + 'n'}
+                    onClick={this.handleOnDateClick}
+                    className="text-center text-secondary"
+                >
+                    {i}
+                </td>
+            );
+        }
+
+        const total = [...prevMonthDays, ...days, ...nextMonthDays];
+
+        let rows = [];
+
+        for (let i = 0; i < total.length; i += 7) {
+            rows.push(total.slice(i, i + 7));
+        }
+
+        const daysInMonth = rows.map((d, i) => {
+            return <tr key={d + i}>{d}</tr>;
+        });
+
+        const { selectedDate } = this.state;
+
+        let years = [];
+        const currentYear = new Date();
+        for (
+            let y = currentYear;
+            y > subYears(currentYear, 20);
```

Comment:
Why hard-code only the last 20 years?

Reply:
I can make constant which can be set to the desired value...